tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1467,6 +1467,59 @@ tensorflow::Status ConvertPlugin(Converter& ctx,
   return tensorflow::Status::OK();
 }
 
+tensorflow::Status ConvertTranspose(
+    Converter& ctx, const tensorflow::NodeDef& node_def,
+    const std::vector<TRT_TensorOrWeights>& inputs,
+    std::vector<TRT_TensorOrWeights>* outputs) {
+  if (inputs.size() != 2 || !inputs.at(0).is_tensor() ||
+      !inputs.at(1).is_weights()) {
+    return tensorflow::errors::InvalidArgument(
+        "Input expects tensor and weights, at", node_def.name());
+  }
+  nvinfer1::ITensor* input_tensor = const_cast<nvinfer1::ITensor*>(
+      inputs.at(0).tensor());
+
+  TRT_ShapedWeights weights = inputs.at(1).weights();
+  const int* weights_ptr = static_cast<int*>(const_cast<void*>(
+      weights.GetValues()));
+  std::vector<int> perm(weights.count());
+  for (int i = 0; i < weights.count(); i++) {
+    perm[i] = weights_ptr[i];
+  }
+
+  nvinfer1::ITensor* output_tensor = ctx.TransposeTensor(input_tensor, perm);
```

Comment:
Sure! Is this new signature okay? `tensorflow::Status TransposeTensor(nvinfer1::ITensor* input_tensor, const std::vector<int>& order, const nvinfer1::ITensor** output_tensor)` Not sure about the const, just mimicking PrepareTensorForShape.

Should I also make a similar change for PrepareTensorForShape to return a status instead of bool? Should PrepareTensorForShape be a member function of Converter like TransposeTensor is?

Reply:
The signature is perfect, and the ideas about `PrepareTensorForShape` are awesome! Thanks.